packages/embark/src/cmd/cmd_controller.js

```diff
@@ -288,7 +290,10 @@ class EmbarkController {
           engine.logger.info(__("loaded plugins") + ": " + pluginList.join(", "));
         }
 
-        if (engine.ipc.connected) {
+        if (isSecondaryProcess(engine)) {
+          engine.startService("web3");
+          engine.startService("deployment");
+          engine.startService("codeGenerator");
           engine.startService("codeRunner");
           engine.startService("console");
```

Comment:
You could move those above the `if` and remove them from below, so we have all the registerService once but in the `if` we return so we don't do those that aren't needed.

Reply:
This has been updated. It was possible due to the events manager oracle (#1340). Without the oracle, services were not able to be reordered as event requests would go unfired if requested before they were handled, causing hangs.